### PR TITLE
E field surf integral non cubic cell test cases

### PR DESCRIPTION
**Pull request by saurabh-s-sawant**
Examples Examples/Tests/Macroscopic_Maxwell/inputs_3d_surfaceIntegralDiagnostic_tests and Examples/Tests/Macroscopic_Maxwell/inputs_3d_surfaceIntegralDiagnostic_tests_cubic have been added to verify the reduced diagnostics for computing surface integral. Tests involve guidance on how to choose a particular plane of integration. It is shown that both nodality and the choice of maximum_grid_size affect the computation fields.  In these tests, only E fields is initialized everywhere with a value of 1. B fields are zero.

### CODE REVIEW

File: Source/Diagnostics/ReducedDiags/MultiReducedDiags.cpp [hunk 1]
```diff
@@ -70,7 +72,7 @@ MultiReducedDiags::MultiReducedDiags ()
             // read reduced diags type
             std::string rd_type;
             pp_rd_name.get("type", rd_type);
-
+            std::cout << "[Debug] rd_type: " <<  rd_type << "\n";
```
Comment (RevathiJambunathan) [inline Source/Diagnostics/ReducedDiags/MultiReducedDiags.cpp]:
```suggestion
```

File: Source/Diagnostics/ReducedDiags/RawEFieldReduction.H [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+                reducedEy_value *= dx[0]*dx[1];
+                reducedEz_value *= dx[0]*dx[1];
+#else
+                reducedEx_value *= dx[0]*dx[1]*dx[2];
+                reducedEy_value *= dx[0]*dx[1]*dx[2];
+                reducedEz_value *= dx[0]*dx[1]*dx[2];
+#endif
+                } else {
+#if (AMREX_SPACEDIM==2)
+		amrex::Real length = m_surface_normal[0]*dx[1] + m_surface_normal[1]*dx[0];
```
Comment (RevathiJambunathan) [inline Source/Diagnostics/ReducedDiags/RawEFieldReduction.H]: can you align the code line here with the ones below

File: Source/Diagnostics/ReducedDiags/RawEFieldReduction.H [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+    {
+        using namespace amrex::literals;
+
+        auto & warpx = WarpX::GetInstance();
+        const auto nLevel = 1;
+        auto reduction_function_parser = m_parser->compile<m_nvars>();
+        int integral_type = m_integral_type;        
+	int* surface_normal = &m_surface_normal[0];
+
+//	amrex:: IntVect surface_normal= amrex::IntVect::TheZeroVector();
```
Comment (RevathiJambunathan) [inline Source/Diagnostics/ReducedDiags/RawEFieldReduction.H]:
```suggestion
```

Comment (RevathiJambunathan) [reply Source/Diagnostics/ReducedDiags/RawEFieldReduction.H]: cleaning commented code

File: Source/Diagnostics/ReducedDiags/RawEFieldReduction.H [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+    template<typename ReduceOp>
+    void ComputeRawEFieldReduction()
+    {
+        using namespace amrex::literals;
+
+        auto & warpx = WarpX::GetInstance();
+        const auto nLevel = 1;
+        auto reduction_function_parser = m_parser->compile<m_nvars>();
+        int integral_type = m_integral_type;        
+	int* surface_normal = &m_surface_normal[0];
```
Comment (RevathiJambunathan) [inline Source/Diagnostics/ReducedDiags/RawEFieldReduction.H]: align code line with the one above

File: Source/Diagnostics/ReducedDiags/RawEFieldReduction.H [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+    /// 3 elements are x, y, z
+    static constexpr int m_nvars = 3;
+    std::unique_ptr<amrex::Parser> m_parser;
+
+    // Type of reduction (e.g. Maximum, Minimum or Sum)
+    int m_reduction_type;
+    // Type of integration (e.g. volume or surface)
+    int m_integral_type;
+#if (AMREX_SPACEDIM==2)
+    int m_surface_normal[2]={0,0};
```
Comment (RevathiJambunathan) [inline Source/Diagnostics/ReducedDiags/RawEFieldReduction.H:68]: Please can you add comments to explain the new integer member variable added

**Review by RevathiJambunathan — CHANGES_REQUESTED** [review]
Thank you for this PR!

Giving this a quick review 
@PeculiarOvertones please can you add the new input parameters in the documentation too.
`Docs/source/usage/parameters.rst`

File: Source/Diagnostics/ReducedDiags/RawEFieldReduction.H (hunk 2 above)
Comment (saurabh-s-sawant) [reply Source/Diagnostics/ReducedDiags/RawEFieldReduction.H]: I did.

File: Source/Diagnostics/ReducedDiags/RawEFieldReduction.H (hunk 4 above)
Comment (saurabh-s-sawant) [reply Source/Diagnostics/ReducedDiags/RawEFieldReduction.H]: I did.

File: Source/Diagnostics/ReducedDiags/RawEFieldReduction.H (hunk 5 above)
Comment (saurabh-s-sawant) [reply Source/Diagnostics/ReducedDiags/RawEFieldReduction.H:68]: I did.

**Pull request closed without merging**